Updated nimbus-cli docs Part II (#454)
mozilla/experimenter-docs: docs/deep-dives/mobile/nimbus-cli/30-opening-and-resetting.mdx

```

#### Deeplinks for other commands:

> iOS `simctl` does not support command line arguments with the `openurl` command. Opening with a deeplink while enrolling in an experiment is not supported.

### `enroll`

For Android, opening a deeplink uses the same mechanism as launching the app: using Intents. Thus, we can enroll in an experiment and then open the app with a deeplink.

```sh
nimbus-cli --app fenix --channel developer enroll release-android-onboarding-redesign --branch treatment-a --deeplink onboarding/privacy-first
```

### `test-feature`

Similarly, you can use a `--deeplink` to enroll a [`test-feature`](/nimbus-cli/working-with-features) experiment.

## Terminating and Resetting the app

`open`, `enroll` and `test-feature` all support a `--reset-app` option, which serves to reset the app back to its just-installed state before opening.

## Passthrough parameters

Opening apps with `nimbus-cli` provides an abstraction over `adb shell start am` and `xcrun simctl launch`.

Passthrough parameters offer a small escape hatch from this abstraction.

### Trailing parameters for apps

> You may want to send additional command line parameters to these commands. You can do this by appending them after `--` for `open`, `enroll` and `test-feature`:

Reviewer: ```suggestion
You may want to send additional command line parameters to these commands. You can do this by appending them with `--` after the `open`, `enroll`, or `test-feature` commands:
```

These are entirely app specific. e.g. Firefox for iOS has the launch parameters available in [`LaunchArguments.swift`](https://github.com/mozilla-mobile/firefox-ios/blob/main/Shared/LaunchArguments.swift).

For Android apps, you can modify the intent being sent to the device with [intent arguments](https://developer.android.com/tools/adb#IntentSpec). The most saliant are reproduced here:

| Extra parameters | Description |
| - | - |
| `--esn extra_key` | Add a null extra. This option is not supported for URI intents. |
| `--es extra_key extra_string_value` | Add string data as a key-value pair. |
| `--ez extra_key extra_boolean_value` |  Add boolean data as a key-value pair. |
| `--ei extra_key extra_int_value` | Add integer data as a key-value pair. |
| `--el extra_key extra_long_value` |  Add long data as a key-value pair. |

For a fictional example:

```sh
nimbus-cli --app fenix --channel developer open --reset-app -- --esn SKIP_INTRO
```

produces an `adb` command:

```sh
$ adb shell "am start -n org.mozilla.fenix.debug/.App \
        -a android.intent.action.MAIN \
        -c android.intent.category.LAUNCHER \
        --esn \
        SKIP_INTRO"
```

### Leading parameters

> For finer control of `xcrun` and `adb`, you can use `{}` as a positional indicator:

Reviewer: 👏
